samtranslator/model/sam_resources.py

```diff
@@ -2194,3 +2202,108 @@ def _construct_appsync_schema(self, api_id: Intrinsicable[str]) -> GraphQLSchema
         schema.DefinitionS3Location = self.SchemaUri
 
         return schema
+
+
+class SamGraphQLDataSource(SamResourceMacro):
+    """SAM GraphQL Data Source Macro (WIP)."""
+
+    # TODO: This currently only supports DataSource with a defined ServiceRoleArn.
+    #       The proposed connector change will come in a follow up PR.
+    resource_type = "AWS::Serverless::GraphQLDataSource"
+    property_types = {
+        "ApiId": Property(True, one_of(IS_STR, IS_DICT)),
+        "Name": Property(False, IS_STR),
+        "Type": Property(True, IS_STR),
+        "Description": PassThroughProperty(False),
+        "ServiceRoleArn": Property(False, IS_STR),
+        "Permissions": Property(False, list_of(IS_STR)),
+        "DataSourceConfig": Property(True, IS_DICT),
+    }
+
+    ApiId: Intrinsicable[str]
+    Type: str
+    DataSourceConfig: DataSourceConfig
+    Name: Optional[str]
+    Description: Optional[PassThrough]
+    ServiceRoleArn: Optional[str]
+
+    @cw_timer
+    def to_cloudformation(self, **kwargs: Any) -> List[Resource]:
+        appsync_datasource = self._construct_appsync_datasource()
+        resources: List[Resource] = [appsync_datasource]
+
+        return resources
+
+    def _validate_dynamodb_datasource(self, ddb_properties: Dict[str, Any]) -> DynamoDBConfig:
+        ddb_config = {}
+
+        ddb_config["TableName"] = sam_expect(
+            ddb_properties.get("TableName"),
+            self.logical_id,
+            "DataSourceConfig.DynamoDB.TableName",
+        ).to_be_a_string()
+
+        ddb_config["AwsRegion"] = ddb_properties["Region"] if "Region" in ddb_properties else ref("AWS::Region")
+
+        if "DeltaSync" in ddb_properties:
+            deltasync_properties = ddb_properties["DeltaSync"]
+
+            sam_expect(
```

Comment:
aren't these all pass through properties? Do we need to run validation on them?

Reply:
They're all required properties if `DeltaSync` is defined by the customer, so I added validation to confirm the fields exist.

Reply:
Discussed offline, these properties will have default values so we need validation incase the user tries to override them